elytron/WFCORE-4750-regex-role-mapper.adoc

```diff
@@ -0,0 +1,159 @@
+
+* In the following example, the original roles:
+
+  - `app-admin`
+  - `batch-admin`
+  - `application-admin`
+  +
+ would be translated into a single `admin` security role, while:
+
+ - `app-operator`
+ - `batch-operator`
+ - `application-operator`
+
+would be translated into a single `operator` security role.
+Roles that don't match the regex won't be preserved.
+
+[source,xml]
+----
+<mappers>
+...
+	<regex-role-mapper name="my-regex-role-mapper" pattern=".*-([a-z]*)$" replacement="$1" keep-non-mapped="false"/>
+...
+</mappers>
+
+----
+
+* The following example demonstrates a use case in which original roles are loaded from a LDAP source in the following format:
+
+ - `APP-123_XY_ZX_ABCD-Batch_Admin`
+ - `APP-ABC_EF_GH_IJKL-Batch_Operator`
+ +
+The mapper below can be used to translate these roles to
+ - `Admin`
+ - `Operator`
+
+[source,xml]
+----
+<regex-role-mapper name="my-rrm" regex=".*_([a-zA-Z]*)$" replacement="$1" />
+----
+
+Note: In order to translate the original roles `Admin` and `Operator` into `admin` and `operator` security roles, with lower case initials, `mapping-role-mapper` can be used and chained on top of this one.
+
+ * In the example below, actual security roles are obtained by extracting the part of the input string preceding the first '_' character and then by converting it to a new format, namely `APP-{extracted_role}`.
+E.g.: the input `USER_ABC_DEF`, `ADMIN_123` roles would be translated into `APP-USER` and `APP-ADMIN` security roles).
+
+
+[source,xml]
+----
+<regex-role-mapper name="my-rrm" regex="^([a-z]*)_.*" replacement="APP-$1" />
+----
+
+
+ * In the example below the domain part is extracted from input roles that are in form of an email address in order to convert them to the `{extracted_domain}-role` format.
+E.g.: `user@gmail.com` and `user@customApp.com` input roles would result in `gmail-role` and `customerApp-role` security roles).
+`keep-non-mapped="true"` will allow to keep the roles that do not match the regex (those that were not in form of an email).
+
+[source,xml]
+----
+<regex-role-mapper name="my-rrm" pattern=".*@([a-z]*)\..*" replacement="$1-role" keep-non-mapped="true"/>
+----
+
+ * The following role mapper can be used to replace all instances of the `guest` substring with the` user` substring.
+E.g. the `abc-guest-abc-guest` input role would be mapped to the` abc-user-abc-user` security role.
+
+[source,xml]
+----
+<regex-role-mapper name="my-rrm" pattern="guest" replacement="user" replace-all="true"/>
+----
+
+Both the pattern and replacement will be in the form of STRING in the model.
+
+
+== Test Plan
+
+* WildFly Elytron test suite: Functional tests  - tests for regex role mapper functionality
+* WildFly Core tess suite: Functional tests - testing of regex role mapper functionality when it is defined in the subsystem
```

Comment:
So the Wildfly Test Suite will not be involved at all?

Reply:
@fabiobrz No the WildFly test suite will not be involved. Should I remove it from affected components to make it clear? Purpose of adding WildFly Security there was because it introduces changes to the Elytron subsystem in the WildFly server, but maybe it is confusing.

Reply:
No, it makes definitely sense to have it there. This was just a question that I asked for clarification. Thanks.